description.md

```diff
@@ -0,0 +1,937 @@
+
+The full list is as follows:
+
+- `fillOrder` - fills a simple order without extensions.
+- `fillOrderExt` - allows the specification of extensions used for the order.
+- `fillOrderTo` - allows the specification of the maker's funds destination instead of `msg.sender`.
+- `fillOrderToExt` - allows the specification of the maker's funds destination and extensions used for the order.
+- `fillOrderToWithPermit` - allows the specification of the maker's funds destination and calls permit before filling the order.
+- `fillContractOrder` - uses contract-based signatures.
+- `fillContractOrderWithPermit` - uses contract-based signatures and the taker's permit.
+- `fillContractOrderExt` - uses contract-based signatures, the taker's permit, and allows the specification of the order's extensions.
+
+All the functions have a similar signature. Here are examples showing the possible arguments and their descriptions:
+
+```solidity
+// Fill from EOA
+function fillOrderToExt(
+        IOrderMixin.Order calldata order,
+        bytes32 r,
+        bytes32 vs,
+        uint256 amount,
+        TakerTraits takerTraits,
+        address target,
+        bytes calldata interaction,
+        bytes calldata extension
+    ) public payable returns(uint256 makingAmount, uint256 takingAmount, bytes32 orderHash) {
+
+// Fill from smart contract
+function fillContractOrderExt(
+        IOrderMixin.Order calldata order,
+        bytes calldata signature,
+        uint256 amount,
+        TakerTraits takerTraits,
+        address target,
+        bytes calldata interaction,
+        bytes calldata permit,
+        bytes calldata extension
+    ) public returns(uint256 makingAmount, uint256 takingAmount, bytes32 orderHash)
+```
+
+| Argument | Type | Description |
+| --- | --- | --- |
+| order | `Order` (calldata) | The order structure to be filled.  |
+| r | `bytes32` | The r-component of the maker’s signature to check that the order hash is signed by the maker. |
+| vs | `bytes32` | The vs-component of the maker’s signature to check that the order hash is signed by the maker. |
+| signature | `bytes calldata` | The signature used to verify the order. It is used for contract-signed orders only. |
+| amount | `uint256` | The amount to fill the order, which can be treated as the maker or taker amount, depending on fill settings. If the amount is greater than the remaining amount to fill, the fill will be executed only for the remaining amount. When partial fills are not allowed, the fill will be reverted if the amount does not equal the order making amount. The fill will also be reverted if the making and taking amounts are equal to zero. |
```

Comment:
Small update about `making and taking amounts are equal to zero`
```
unchecked { if (makingAmount * takingAmount == 0) revert SwapWithZeroAmount(); }
```

Reply:
fixed